### PR TITLE
[12.0][MIG] Migrate l10n_ch_fds_postfinance

### PR DESCRIPTION
**Pull request by philhug**

### CODE REVIEW

**Comment by philhug** [comment]
where can I see the runbot logs? I get a 404 when trying to access them

**Comment by yvaucher** [comment]
@philhug runbot gives a failing state even if it runs but has warnings

It is worth it to avoid unnecessary warnings as those could generate lots of lines in your logs thus bloating your kibana or other tool.
Looking at travis logs I think you need to fix those warnings to get a green runbot:

```
2020-04-23 11:32:43,603 5838 WARNING openerp_test odoo.models: The model fds.authentication.keys has no _description
2020-04-23 11:32:43,604 5838 WARNING openerp_test odoo.models: The model fds.postfinance.account has no _description
2020-04-23 11:32:43,604 5838 WARNING openerp_test odoo.models: The model fds.postfinance.file has no _description
2020-04-23 11:32:43,604 5838 WARNING openerp_test odoo.models: The model fds.postfinance.directory has no _description
2020-04-23 11:32:43,605 5838 WARNING openerp_test odoo.models: The model fds.files.import.tobankstatments.wizard has no _description
2020-04-23 11:32:43,605 5838 WARNING openerp_test odoo.models: The model fds.key.clone.wizard has no _description
2020-04-23 11:32:43,606 5838 WARNING openerp_test odoo.models: The model fds.key.generator.wizard has no _description
2020-04-23 11:32:43,845 5838 INFO openerp_test odoo.modules.registry: module l10n_ch_fds_postfinance: creating or updating database tables
2020-04-23 11:32:44,001 5838 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (public_key_import_file, public_key_import_txt) of fds.key.import.wizard() have the same label: Public key.
2020-04-23 11:32:44,001 5838 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (private_key_import_file, private_key_import_txt) of fds.key.import.wizard() have the same label: Private key.
2020-04-23 11:32:44,087 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/data/import_files_cron.xml
2020-04-23 11:32:44,106 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_authentication_keys_view.xml
2020-04-23 11:32:44,128 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_files_import_tobankstatments_wizard_view.xml
2020-04-23 11:32:44,164 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_key_clone_wizard_view.xml
2020-04-23 11:32:44,182 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_key_generator_wizard_view.xml
2020-04-23 11:32:44,201 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_key_import_wizard_view.xml
2020-04-23 11:32:44,222 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_postfinance_account_view.xml
2020-04-23 11:32:44,276 5838 INFO openerp_test odoo.modules.loading: loading l10n_ch_fds_postfinance/views/fds_postfinance_file_view.xml
2020-04-23 11:32:44,342 5838 WARNING openerp_test odoo.models: ir.actions.server.create() with unknown fields: key2, value
2020-04-23 11:32:44,357 5838 WARNING openerp_test odoo.models: ir.actions.server.create() with unknown fields: key2, value
2020-04-23 11:32:44,373 5838 WARNING openerp_test odoo.models: ir.actions.server.create() with unknown fields: key2, value
```

File: l10n_ch_fds_postfinance/views/fds_postfinance_file_view.xml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    <record id="file_cancel" model="ir.actions.server">
+        <field name="name">Cancel</field>
+        <field name="model_id" ref="model_fds_postfinance_file"/>
+        <field name="state">code</field>
+        <field name="code">records.change2cancel_button()</field>
+    </record>
+
+    <record id="file_cancel_more_item" model="ir.actions.server">
+        <field eval="'client_action_multi'" name="key2"/>
```
Comment (yvaucher) [inline l10n_ch_fds_postfinance/views/fds_postfinance_file_view.xml:115]: this raises warning, please check this key is still valid.

File: l10n_ch_fds_postfinance/wizards/fds_files_import_tobankstatments_wizard.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    )
+
+
+class FdsFilesImportToBankStatementsWizard(models.TransientModel):
+    """ This wizard checks and downloads files in FDS Postfinance server
+        that were not already downloaded on the database.
+        This wizard is called when we choose the update_fds for one FDS.
+    """
+    _name = 'fds.files.import.tobankstatments.wizard'
+
```
Comment (yvaucher) [inline l10n_ch_fds_postfinance/wizards/fds_files_import_tobankstatments_wizard.py:33]: `_description` to add on models.

File: l10n_ch_fds_postfinance/wizards/fds_key_import_wizard.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    private_key_import_txt = fields.Text(
+        string='Private key',
+        help='copy/paste your private key'
+    )
+    public_key_import_file = fields.Binary(
+        string='Public key',
+        help='select one file that contain your public key'
+    )
+    private_key_import_file = fields.Binary(
+        string='Private key',
```
Comment (yvaucher) [inline l10n_ch_fds_postfinance/wizards/fds_key_import_wizard.py:35]: Warning your `string` are the same for:
- public_key_import_txt and public_key_import_file
- private_key_import_txt and private_key_import_file

**Review by yvaucher — CHANGES_REQUESTED** [review]
To solve the warnings here are few pointers

**Comment by vvrossem** [comment]
Hi @philhug :) Thank your for this PR!
I'm trying to install the module for testing purposes on a database created with `Language:French (CH)` and `Country:Switzerland`.

Unfortunately, the following error is raised when loading the `fr_CH.po ` (and is not if database is created with French or English as language): 
```
(...)
2020-10-21 15:24:13,426 128743 INFO twelve-l10n_ch_fds_postfinance odoo.addons.base.models.ir_translation: module l10n_ch_fds_postfinance: loading translation file (fr_CH) for language fr_CH 
2020-10-21 15:24:13,426 128743 INFO twelve-l10n_ch_fds_postfinance odoo.tools.translate: loading /home/vincent/twelve-odoo/oca/l10n-switzerland/l10n_ch_fds_postfinance/i18n/fr_CH.po 
2020-10-21 15:24:13,445 128743 ERROR twelve-l10n_ch_fds_postfinance odoo.sql_db: bad query:  INSERT INTO ir_translation(name, lang, res_id, src, type, value, module, state, comments)
                           SELECT name, lang, res_id, src, type, value, module, state, comments
                           FROM tmp_ir_translation_import
                           WHERE type = 'model'
                           AND noupdate IS NOT TRUE
                           ON CONFLICT (type, lang, name, res_id) WHERE type = 'model'
                            DO UPDATE SET (name, lang, res_id, src, type, value, module, state, comments) = (EXCLUDED.name, EXCLUDED.lang, EXCLUDED.res_id, EXCLUDED.src, EXCLUDED.type, EXCLUDED.value, EXCLUDED.module, EXCLUDED.state, EXCLUDED.comments)
                            WHERE EXCLUDED.value IS NOT NULL AND EXCLUDED.value != '';
                       
ERROR: ON CONFLICT DO UPDATE command cannot affect row a second time
HINT:  Ensure that no rows proposed for insertion within the same command have duplicate constrained values.
```

From what I'm reading [here](https://www.odoo.com/forum/help-1/question/on-conflict-do-update-problem-141853), it could be that a word is translated twice in `.po` file. I've looked the into the `fr_CH.po` and I haven't found any.
For a reason I cannot explain yet, the following `msgid` should not be translated to prevent raising the error:
- "Active"
- "Cancel"
- "Close"
- "Generate"

**Comment by yvaucher** [comment]
Closed in favor of #570

**Pull request closed without merging**